include/battle.h

```diff
@@ -601,6 +601,12 @@ struct BattleGimmickData
     bool8 activated[MAX_BATTLERS_COUNT][GIMMICKS_COUNT]; // stores whether a trainer has used gimmick
 };
 
+struct SleepClauseData
+{
+    bool8 effectExempt; // Stores whether effect should be exempt from triggering Sleep Clause (Effect Spore)
+    u8 monCausingSleepClause[NUM_BATTLE_SIDES]; // Stores which pokemon on a given side is causing Sleep Clause to be active as the mon's index in the party
+};
```

Comment:
Could this be just moved to the Battle Struct instead? `effectExempt` only needs 1 bit so 7 bits are wasted. This is minor but the optimization here is free. `effectExempt` could be renamed to something like sleepClauseEffectExempt. If you want them togher just add some padding under `sleepClauseEffectExempt`, so:
```
u8 monCausingSleepClause[NUM_BATTLE_SIDES];
u8 sleepClauseEffectExempt:1;
u8 paddingX:7;

Reply:
I'm on board for that, I would want them to stay together. I default to new structs because I find it more organized but I get that on our target platform we should take every optimization we can get. :)